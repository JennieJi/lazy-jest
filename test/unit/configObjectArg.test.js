const configObjectArg = require('../../src/caseGenerator/configObjectArg');
const configArgs = require('../../src/caseGenerator/configArgs');

describe('configObjectArg', () => {
  test('empty object', () => {
    expect(configObjectArg(configArgs())).toMatchSnapshot();
  });

  test('single property', () => {
    expect(configObjectArg(configArgs().arg('a', [1]))).toMatchSnapshot();
  });

  test('single property has invalid cases', () => {
    expect(configObjectArg(configArgs().arg('a', [1], { invalidCases: [null] }))).toMatchSnapshot();
  });

  test('single optional property', () => {
    expect(configObjectArg(configArgs().arg('a', [1], { optional: true }))).toMatchSnapshot();
  });

  test('optional property has invalid cases', () => {
    expect(configObjectArg(configArgs().arg('a', [1], { optional: true, invalidCases: [null] }))).toMatchSnapshot();
  });

  test('multiple properties, no optional', () => {
    expect(configObjectArg(
      configArgs()
      .arg('a', [1])
      .arg('b', [0])
    )).toMatchSnapshot();
  });

  test('multiple properties, has optional', () => {
    expect(configObjectArg(
      configArgs()
      .arg('a', [1])
      .arg('b', [0], { optional: true })
    )).toMatchSnapshot();
  });
  
  test('multiple properties, has invalid cases', () => {
    expect(configObjectArg(
      configArgs()
      .arg('a', [1])
      .arg('b', [0], { invalidCases: [null] })
    )).toMatchSnapshot();
  });

  test('multiple properties in array', () => {
    expect(configObjectArg([
      { name: 'a', validCases: [1] },
      { name: 'b', validCases: [0] }
    ])).toMatchSnapshot();
  });
});