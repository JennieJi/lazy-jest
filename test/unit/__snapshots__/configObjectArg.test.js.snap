// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configObjectArg empty object 1`] = `
Object {
  "invalidCases": Array [],
  "validCases": Array [],
}
`;

exports[`configObjectArg multiple properties in array 1`] = `
Object {
  "invalidCases": Array [],
  "validCases": Array [
    Object {
      "a": 0,
    },
    Object {
      "b": 0,
    },
  ],
}
`;

exports[`configObjectArg multiple properties, has invalid cases 1`] = `
Object {
  "invalidCases": Array [
    Object {
      "b": null,
    },
  ],
  "validCases": Array [
    Object {
      "a": 0,
    },
    Object {
      "b": 0,
    },
  ],
}
`;

exports[`configObjectArg multiple properties, has optional 1`] = `
Object {
  "invalidCases": Array [],
  "validCases": Array [
    Object {
      "a": 1,
    },
    Object {
      "a": 1,
      "b": 0,
    },
    Object {
      "a": 1,
      "b": undefined,
    },
  ],
}
`;

exports[`configObjectArg multiple properties, no optional 1`] = `
Object {
  "invalidCases": Array [],
  "validCases": Array [
    Object {
      "a": 0,
    },
    Object {
      "b": 0,
    },
  ],
}
`;

exports[`configObjectArg optional property has invalid cases 1`] = `
Object {
  "invalidCases": Array [
    Object {
      "a": null,
    },
  ],
  "validCases": Array [
    Object {
      "a": 1,
    },
    Object {
      "a": undefined,
    },
  ],
}
`;

exports[`configObjectArg single optional property 1`] = `
Object {
  "invalidCases": Array [],
  "validCases": Array [
    Object {
      "a": 1,
    },
    Object {
      "a": undefined,
    },
  ],
}
`;

exports[`configObjectArg single property 1`] = `
Object {
  "invalidCases": Array [],
  "validCases": Array [
    Object {
      "a": 1,
    },
  ],
}
`;

exports[`configObjectArg single property has invalid cases 1`] = `
Object {
  "invalidCases": Array [
    Object {
      "a": null,
    },
  ],
  "validCases": Array [
    Object {
      "a": 1,
    },
  ],
}
`;
